alchemist/alchemist-ns3/src/main/java/it/unibo/alchemist/ns3/utils/DefaultNs3Serializer.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2010-2020, Danilo Pianini and contributors
+ * listed in the main project's alchemist/build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
+package it.unibo.alchemist.ns3.utils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.OutputStream;
+
+/**
+ * A {@link Serializer} which uses a {@link ObjectOutputStream} to serialize objects and
+ * a {@link ObjectInputStream} to deserialize them.
+ */
+public final class DefaultNs3Serializer implements Serializer {
```

Comment:
Is this bound statically? Otherwise, needs to be serializable.

Reply:
At the moment the serializer is only set when the simulation is built from the yml file

Reply:
What if the environment is serialized and then deserialized?
I believe using this class would break serialization